charts/sqnc-node/templates/tests/checkIfValidator.yaml

```diff
@@ -0,0 +1,103 @@
+{{ $fullname :=  include "sqnc-node.fullname" . }}
+{{ $serviceLabels :=  include "sqnc-node.serviceLabels" .  }}
+{{ $selectorLabels :=  include "sqnc-node.selectorLabels" .  }}
+
+{{range $i := until ($.Values.node.replicas | int) }}
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ $fullname }}-{{ $i }}-post-install-test-suite"
+  labels:
+      {{- $serviceLabels | nindent 4 }}
+  annotations:
+    "helm.sh/hook": post-install
+spec:
+  containers:
+    {{- if eq $.Values.node.role "validator" }}
+    - name: block-authoring-check
+      image: {{ $.Values.osShell.image.repository }}:{{ $.Values.osShell.image.tag }}
+      command: [ "/bin/sh" ]
+      args:
+        - -c
+        - |
+          START_HEAD=`curl -sS -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "chain_getFinalizedHead", "params": []}' "http://{{ $fullname }}-{{ $i }}:9944" | jq '.result'`
+
+          # Authoring check
+          START_BLOCK_LOGS=`curl -sS -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "chain_getHeader", "params": ['$START_HEAD']}' "http://{{ $fullname }}-{{ $i }}:9944" | jq '.result.digest.logs'`
+          START_BLOCK_AUTHOR=$(echo "$START_BLOCK_LOGS" | cut -d 'x' -f 2 | sed -r 's/(.{2})/\\x&/g')
+          if [ "${START_BLOCK_AUTHOR##*"BABE"*}" ]; then
+            echo "caught BABE as author for block $START_HEAD"
+          else
+            echo "failed to match expected block author"
+            exit
+          fi
+
+          # Finalisation check
+          CURRENT_HEAD=$START_HEAD
+
+          echo "block head at start: $START_HEAD"
+
+          count=$((0)); interval=5; timeout=120
+          while [ $CURRENT_HEAD = $START_HEAD ]; do
+            sleep "$interval"s
+
+            CURRENT_HEAD=`curl -sS -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "chain_getFinalizedHead", "params": []}' "http://{{ $fullname }}-{{ $i }}:9944" | jq '.result'`
+
+            echo "block head after ${interval}s: $CURRENT_HEAD"
+
+            count=$((count+interval))
+
+            if [ $count -ge $timeout ]; then
+              echo "block finalisation timeout after $count seconds; no change detected"
+              break
+            else
+              echo "block finalisation witnessed after $count seconds"
+            fi
+          done
+    {{- end }}
+    - name: block-height-check
+      image: {{ $.Values.osShell.image.repository }}:{{ $.Values.osShell.image.tag }}
+      command: [ "/bin/sh" ]
+      args:
+        - -c
+        - |
+          SYNC_STATE=`curl -sS -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "system_syncState", "params": []}' "http://{{ $fullname }}-{{ $i }}:9944" | jq '.result'`
```

Comment:
the logic of this check seems sound. The only thing I'd suggest is this check should be delayed by say 30 seconds before performing this call, just to make sure we've had an opportunity to listen for new blocks.